coding/js.md

```diff
@@ -0,0 +1,19 @@
+# JS code standards
+This document outlines the rules for writing Javascript documents and fragments across all of our codebases.
+
+## General rules
+- Javascript should progressively enhance a page. Core content and functionality should still be available without Javascript.
+- Variable names should be camelCase
+- Avoid abbreviated or deliberately short names (choose `query` over `q`).
+- Your code should be sufficiently eloquent that it doesn't need comments to describe its purpose, however, if comments are required, they should be inline.
```

Comment:
I love the idea of talking about expressive code, but could we remove the "that it doesn't need comments" bit? My experience is that people mostly over-anticipate the extent to which their code is self-explanatory, and so we should encourage more information rather than less.